pkgs/by-name/pi/pike/package.nix

```
@@ -0,0 +1,196 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchurl,
+  pkg-config,
+  autoconf,
+  automake,
+  libtool,
+  bison,
+  flex,
+  gmp,
+  pcre,
+  nettle,
+  libjpeg,
+  libpng,
+  libtiff,
+  freetype,
+  sqlite,
+  libmysqlclient,
+  postgresql,
+  openssl,
+  zlib,
+  bzip2,
+  gdbm,
+  libffi,
+  expat,
+  libxml2,
+  libxslt,
+  gtk3,
+  pango,
+  cairo,
+  glib,
+  SDL2,
+  libGL,
+  dpkg,
+  autoPatchelfHook,
+  ncurses,
+  libnsl,
+  libxcrypt-legacy,
+  unstableGitUpdater,
+}:
+
+let
+  # Pre-built Pike binary from Debian for bootstrapping
+  pike-bootstrap = stdenv.mkDerivation rec {
+    pname = "pike-bootstrap";
+    version = "8.0.1956";
+
+    src = fetchurl {
+      # Using Debian's pike8.0-core package
+      url = "http://deb.debian.org/debian/pool/main/p/pike8.0/pike8.0-core_${version}-1_amd64.deb";
+      sha256 = "sha256-kDaP7GZOW9Wlzg9dSxi+Q8IYduIZneWSsFqRfNh1NFQ=";
+    };
+
+    nativeBuildInputs = [
+      autoPatchelfHook
+      dpkg
+    ];
+
+    buildInputs = [
+      gmp
+      zlib
+      ncurses
+      libxcrypt-legacy # for libcrypt.so.1
+      nettle # for libnettle and libhogweed
+      libnsl # for libnsl.so
+    ];
+
+    # Ignore missing dependencies since this is just for bootstrapping
+    autoPatchelfIgnoreMissingDeps = [
+      "libcrypt.so.1"
+      "libnsl.so.2"
+    ];
+
+    unpackPhase = ''
+      dpkg-deb -x $src .
+    '';
```

You shouldn't have to set this explictly since there is a setuphook
https://github.com/NixOS/nixpkgs/blob/ba9d059a2b06fd6ea9393bdf106a85cded11e0d5/pkgs/by-name/dp/dpkg/setup-hook.sh

And if for some reason you do
```suggestion
    unpackCmd = "dpkg -x $src .";
```